### PR TITLE
CRUD added for Treatment + Medicine (not complete)

### PR DESCRIPTION
**Pull request by anamariaghiorghiu**

### CODE REVIEW

File: src/DucksNet.API/Controllers/TreatmentController.cs [hunk 1]
```diff
@@ -9,9 +11,11 @@ namespace DucksNet.API.Controllers;
 public class TreatmentController : ControllerBase
 {
     private readonly IRepository<Treatment> _treatmentRepository;
-    public TreatmentController(IRepository<Treatment> repository)
+    //private readonly IRepository<Medicine> _medicineRepository;
```
Comment (OpariucRares) [inline src/DucksNet.API/Controllers/TreatmentController.cs]: mark as a TODO (abbreviation of your name)

File: src/DucksNet.Domain/Model/Medicine.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿using DucksNet.SharedKernel.Utils;
 using DucksNet.Domain.Model.Enums;
+using System.Net;
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/Medicine.cs]: does it need?

File: tests/DucksNet.IntegrationTests/TreatmentControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿//using System;
+//using System.Collections.Generic;
+//using System.Linq;
+//using System.Text;
+//using System.Threading.Tasks;
+
+//namespace DucksNet.IntegrationTests;
```
Comment (OpariucRares) [inline tests/DucksNet.IntegrationTests/TreatmentControllerTests.cs]: why is it commented? make at least one test

File: tests/DucksNet.UnitTests/TreatmentTests.cs [hunk 4]
```diff
@@ -1,18 +1,34 @@
...
+    //[Fact]
+    //public void When_CreateTreatments_WithNoMedicine_Should_Fail()
+    //{
+    //    List<Guid> medicineIDList = new List<Guid>();
+    //    var result = Treatment.CreateTreatment(medicineIDList);
+    //    result.IsFailure.Should().BeTrue();
+    //    result.Value.Should().BeNull();
+    //    result.Errors.Should().NotBeEmpty();
+    //    result.Errors.Should().Contain("Can't create treatment with no medicine.");
+    //}
```
Comment (OpariucRares) [inline tests/DucksNet.UnitTests/TreatmentTests.cs]: same here. At least make the treatment domain 100%

**Review by OpariucRares — CHANGES_REQUESTED** [review]
don't add the DucksNet.db, DucksNet.db-shm, DucksNet.db-wal. remove for the push, but keet it local

File: src/DucksNet.Domain/Model/Medicine.cs [hunk 5]
```diff
@@ -21,14 +26,63 @@ private Medicine(string name, string description, double price, DrugAdministrati
...
+        return Result<Medicine>.Ok(new Medicine(name, description, price, typeOfDrugAdministrationByString.Value));
+    }
+
+    public Result UpdateMedicineFields(string name, string description, double price, string drugAdministrationString)
+    {
+        if (!string.IsNullOrWhiteSpace(name))
+        {
+            Name = name;
+        }
+        else if (string.IsNullOrWhiteSpace(name))
```
Comment (OpariucRares) [inline src/DucksNet.Domain/Model/Medicine.cs:58]: you can remove the ifs; let alone else

**Review by OpariucRares — CHANGES_REQUESTED** [review]
make for all the ifs

File: src/DucksNet.Domain/Model/Medicine.cs [hunk 6]
```diff
@@ -21,14 +26,63 @@ private Medicine(string name, string description, double price, DrugAdministrati
...
+            return Result.Error("Description is empty.");
+        }
+        if (price >= 0)
+        {
+            Price = price;
+        }
+        else if (price <= 0)
+        {
+            return Result.Error("Invalid price.");
+        }
```
Comment (Al-Andrew) [inline src/DucksNet.Domain/Model/Medicine.cs:77]: the if and elif overlap on price = 0 here

File: tests/DucksNet.IntegrationTests/BaseIntegrationTests.cs [hunk 7]
```diff
@@ -34,7 +34,7 @@ protected void ClearDatabase()
         context.RemoveRange(context.Users);
         context.RemoveRange(context.MedicalRecords);
         context.RemoveRange(context.Employees);
-        context.RemoveRange(context.Treatments);
+        //context.RemoveRange(context.Treatments);
```
Comment (Al-Andrew) [inline tests/DucksNet.IntegrationTests/BaseIntegrationTests.cs]: Why should we not clean out the Treatments?

File: tests/DucksNet.IntegrationTests/MedicineControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+    //{
+    //    ClearDatabase();
+    //    //Arrange
+    //    Guid id = Guid.NewGuid();
+    //    Medicine medicine = new Medicine(id, "Naldorex" , "Migraine", -999, DrugAdministration.Oral);
+    //    //Act
+    //    var medicineResponse = await TestingClient.PostAsJsonAsync(MedicineUrl, medicine);
+    //    //Assert
+    //    medicineResponse.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+    //}
```
Comment (Al-Andrew) [inline tests/DucksNet.IntegrationTests/MedicineControllerTests.cs]: Why is this test commented out?

**Review by Al-Andrew — APPROVED** [review]
All in all seems fine. But please uncomment out the removing of treatments fro the database when testing.

**Review by OpariucRares — APPROVED** [review]

**Pull request merged**